### PR TITLE
Rename Authchain to Intchain 

### PR DESCRIPTION
**Pull request by Druelik**
# Description of change
renamed authchain to Intchain
## Links to any relevant issues
fixes issue #195 partly 
like discussed in the weekly meeting, everything was renamed but the logic behind it wasnt changed.

## Type of change
Add an `x` to the boxes that are relevant to your changes.

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [X] I have followed the contribution guidelines for this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation

### CODE REVIEW

File: identity-iota/src/chain/document.rs [hunk 1]
```diff
@@ -40,43 +40,43 @@ impl DocumentChain {
...
   pub fn id(&self) -> &DID {
-    self.auth_chain.current.id()
+    self.int_chain.current.id()
   }
 
-  /// Returns a reference to the `AuthChain`.
-  pub fn auth(&self) -> &AuthChain {
-    &self.auth_chain
+  /// Returns a reference to the `IntChain`.
+  pub fn auth(&self) -> &IntChain {
```
Comment (JelleMillenaar) [inline identity-iota/src/chain/document.rs:73]: rename the func?

File: identity-iota/src/chain/document.rs [hunk 2]
```diff
@@ -91,34 +91,34 @@ impl DocumentChain {
...
     if let Some(document) = self.document.as_mut() {
       document
     } else {
-      self.auth_chain.current_mut()
+      self.int_chain.current_mut()
     }
   }
 
   /// Returns the Tangle message Id of the latest auth document.
   pub fn auth_message_id(&self) -> &MessageId {
```
Comment (JelleMillenaar) [inline identity-iota/src/chain/document.rs:115]: rename ;D

File: identity-iota/src/chain/document.rs [hunk 3]
```diff
@@ -127,7 +127,7 @@ impl DocumentChain {
   ///
   /// Fails if the document is not a valid auth document.
   pub fn try_push_auth(&mut self, document: Document) -> Result<()> {
```
Comment (JelleMillenaar) [inline identity-iota/src/chain/document.rs:129]: Rename ;D

**Comment by l1h3r** [comment]
every time I see `IntChain` I think of a chain of integers....

`int` is also a bit of a confusing variable name here (thankfully rust doesn't have the keyword `int`). maybe it would be less confusing to use `IntegrationChain` and not use the `int` variable name ?

**Comment by Druelik** [comment]
done in #230

**Pull request closed without merging**